depends/class-2.6.0-a_max.patch

```diff
@@ -68,3 +68,98 @@ diff -uNr class_public-2.6.0.old/source/input.c class_public-2.6.0/source/input.
+@@ -2279,6 +2279,7 @@
+   double chi_heat;
+   int last_index_back;
+   double relative_variation;
++  double zend=1./pba->a_max - 1;
+
+   Yp = pth->YHe;
+
+@@ -2367,7 +2368,7 @@
+   /* initial step */
+   dz = dz_max;
+
+-  while (z > 0.) {
++  while (z > zend) {
+
+     class_test(dz < ppr->smallest_allowed_variation,
+                pth->error_message,
+@@ -2375,7 +2376,7 @@
+
+     /* - try next step */
+     z_next=z-dz;
+-    if (z_next < 0.) z_next=0.;
++    if (z_next < zend) z_next=zend;
+
+     class_call(thermodynamics_reionization_function(z_next,pth,preio,&xe_next),
+                pth->error_message,
+@@ -2660,7 +2661,7 @@
+   param.nH0 = 11.223846333047*param.obh2*(1.-param.Y);  /* number density of hydrogen today in m-3 */
+   param.fHe = param.Y/(1-param.Y)/3.97153;              /* abundance of helium by number */
+   param.zstart = ppr->recfast_z_initial; /* Redshift range */
+-  param.zend = 0.;
++  param.zend = 1./pba->a_max - 1.;
+   param.dlna = 8.49e-5;
+   param.nz = (long) floor(2+log((1.+param.zstart)/(1.+param.zend))/param.dlna);
+   param.annihilation = pth->annihilation;
+@@ -2799,7 +2800,7 @@
+
+     /** - --> get redshift, corresponding results from hyrec, and background quantities */
+
+-    z = param.zstart * (1. - (double)(i+1) / (double)Nz);
++    z = param.zstart  - (param.zstart - param.zend) * (double)(i+1) / (double)Nz;
+
+     /* get (xe,Tm) by interpolating in pre-computed tables */
+
+@@ -2945,7 +2946,7 @@
+   double y[3],dy[3];
+
+   /* other recfast variables */
+-  double OmegaB,zinitial,x_He0,x0;
++  double OmegaB,zinitial,zfinal,x_He0,x0;
+   double x_H0=0.;
+   double z,mu_H,Lalpha,Lalpha_He,DeltaB,DeltaB_He;
+   double zstart,zend,rhs;
+@@ -2989,6 +2990,9 @@
+   /* z_initial */
+   zinitial=ppr->recfast_z_initial;
+
++  /* z_final */
++  zfinal=1./pba->a_max - 1.;
++
+   /* H_frac */
+   preco->H_frac = ppr->recfast_H_frac;
+
+@@ -3068,8 +3072,8 @@
+
+   for(i=0; i <Nz; i++) {
+
+-    zstart = zinitial * (double)(Nz-i) / (double)Nz;
+-    zend   = zinitial * (double)(Nz-i-1) / (double)Nz;
++    zstart = zinitial  - (zinitial-zfinal) * (double)(i) / (double)Nz;
++    zend   = zinitial  - (zinitial-zfinal) * (double)(i+1) / (double)Nz;
+
+     z = zend;
+
+@@ -3226,7 +3230,7 @@
+
+       class_call(generic_integrator(thermodynamics_derivs_with_recfast,
+                                     zstart,
+-                                    zend,
++                                    0.99999*zend,
+                                     y,
+                                     &tpaw,
+                                     ppr->tol_thermo_integration,
+@@ -3264,7 +3268,7 @@
+     *(preco->recombination_table+(Nz-i-1)*preco->re_size+preco->index_re_Tb)=y[2];
+
+     /* get dTb/dz=dy[2] */
+-    class_call(thermodynamics_derivs_with_recfast(zend, y, dy, &tpaw,pth->error_message),
++    class_call(thermodynamics_derivs_with_recfast(0.99999*zend, y, dy, &tpaw,pth->error_message),
```

Comment:
Otherwise the spline interpolation seems to fail when checking the bounds due to a round off error